numpy/core/src/multiarray/ctors.c

```diff
@@ -825,11 +825,9 @@ PyArray_NewFromDescr_int(
          * Allocate something even for zero-space arrays
          * e.g. shape=(0,) -- otherwise buffer exposure
          * (a.data) doesn't work as it should.
-         * Could probably just allocate a few bytes here. -- Chuck
-         * Note: always sync this with calls to PyDataMem_UserFREE
          */
         if (nbytes == 0) {
-            nbytes = descr->elsize ? descr->elsize : 1;
```

Comment:
Can we ensure all the strides are set to zero to avoid any concerns of undefined behavior?

Reply:
OK, but `array_strides_set` will allow resetting the strides to any values.

Reply:
I have the following behavior when a 0-size array is reshaped in NumPy 1.23:
```
>>> import numpy as np
>>> a = np.ones((2, 0, 3))
>>> a.strides
(0, 0, 0)
>>> b = a.reshape((3, 0, 2))
>>> b.strides
(16, 16, 8)
```
Do you think `reshape` should be also fixed to return `b` with strides `(0, 0, 0)` here, @eric-wieser @mattip ? Thanks!

ref. https://github.com/cupy/cupy/pull/6807

Reply:
That does look like a bug. Please open a new issue for this, the PR here was merged and further discussion will get lost.

Reply:
Thanks! I opened #21918.